rocsolver/library/include/rocsolver-functions.h

```diff
@@ -5740,6 +5740,480 @@ ROCSOLVER_EXPORT rocblas_status rocsolver_zpotrf_strided_batched(
+    rocblas_handle handle, const rocblas_svect left_svect,
+    const rocblas_svect right_svect, const rocblas_int m, const rocblas_int n,
+    rocblas_float_complex *A, const rocblas_int lda, float *S,
+    rocblas_float_complex *U, const rocblas_int ldu, rocblas_float_complex *V,
+    const rocblas_int ldv, float *E, const bool fast_alg, rocblas_int *info);
+
+ROCSOLVER_EXPORT rocblas_status rocsolver_zgesvd(
+    rocblas_handle handle, const rocblas_svect left_svect,
+    const rocblas_svect right_svect, const rocblas_int m, const rocblas_int n,
+    rocblas_double_complex *A, const rocblas_int lda, double *S,
+    rocblas_double_complex *U, const rocblas_int ldu, rocblas_double_complex *V,
+    const rocblas_int ldv, double *E, const bool fast_alg, rocblas_int *info);
+
+/*! \brief GESVD_BATCHED computes the Singular Values and optionally the
+   Singular Vectors of a batch of general m-by-n matrix A (Singular Value
+   Decomposition).
+
+    \details
+    The SVD of matrix A_j is given by:
+
+        A_j = U_j * S_j * V_j'
+
+    where m-by-n matrix S_j is zero except, possibly, for its min(m,n) diagonal
+   elements which are the singular values of A_j. U_j and V_j are orthogonal
+   (unitary) matrices. The first min(m,n) columns of U_j and V_j are the left
+   and right singular vectors of A_j respectively.
+
+    The computation of the singular vectors is optional and it is controlled by
+   the function arguments left_svect and right_svect as described below. When
+   computed, this function returns the tranpose (or transpose conjugate) of the
+   right singular vectors, ie. the rows of V_j'.
+
+    left_svect and right_svect are rocblas_svect enums that can take the
+   following values:
+
+        rocblas_svect_all: the entire matrix U_j (or V_j') is computed,
+        rocblas_svect_singular: only the singular vectors (first min(m,n)
+   columns of U_j or rows of V_j') are computed, rocblas_svect_overwrite: the
+   first columns (or rows) of A_j are overwritten with the singular vectors, or
+        rocblas_svect_none: no columns (or rows) of U_j (or V_j') are computed,
+   ie. no singular vectors.
+
+    left_svect and right_svect cannot both be set to overwrite. When none is set
+   to overwrite, the contents of A_j are destroyed by the time the function
+   returns.
+
+    (When m >> n or (n >> m) the algorithm could be speeded up by compressing
+   the matrix A_j via a QR (or LQ) factorization, and working with the
+    triangular factor afterwards (thin-SVD). If the singular vectors are also
+   requested, its computation could be speeded up as well via executing some
+   intermediate operations out-of-place, and relying more on matrix
+   multiplications (GEMMs); this will require, however, a larger memory
+    workspace. The boolean fast_alg controls whether the fast algorithm is
+   executed or not. For more details see the sections "Tuning rocSOLVER
+   performance" and "Memory model" on the User's guide).
```

Comment:
A lot of my comments are duplicated across these functions. I'll put this one here to be a little less cluttered.
```suggestion
    \note
    When m >> n or (n >> m) the algorithm could be speeded up by compressing
   the matrix A_j via a QR (or LQ) factorization, and working with the
    triangular factor afterwards (thin-SVD). If the singular vectors are also
   requested, its computation could be speeded up as well via executing some
   intermediate operations out-of-place, and relying more on matrix
   multiplications (GEMMs); this will require, however, a larger memory
    workspace. The boolean fast_alg controls whether the fast algorithm is
   executed or not. For more details see the sections "Tuning rocSOLVER
   performance" and "Memory model" in the User's guide.
```
I don't know if you considered the Doxygen `\note` command, but it might be a good format for this sort of aside.

Reply:
Yeah, why not. Changed...